cpp/src/io/parquet/compact_protocol_reader.cpp

```diff
@@ -140,6 +140,7 @@ struct parquet_field_bool_list : public parquet_field_list<bool, FieldType::BOOL
     auto const read_value = [&val = v](uint32_t i, CompactProtocolReader* cpr) {
       auto const current_byte = cpr->getb();
       assert_bool_field_type(current_byte);
+      CUDF_EXPECTS(i < val.size());
```

Comment:
I'm 👍.  But would there be value in also adding a string error message to the `CUDF_EXPECTS`?

Reply:
Done in cbdf13417a5d032b51fb8ac711f3d0dbef7ad61f.